superset/views/dashboard/api.py

```diff
@@ -88,12 +88,12 @@ class BaseDashboardSchema(BaseOwnedSchema):
     @pre_load
     def pre_load(self, data):  # pylint: disable=no-self-use
         super().pre_load(data)
-        data["slug"] = data.get("slug")
-        data["owners"] = data.get("owners", [])
-        if data["slug"]:
+        if data.get("slug"):
             data["slug"] = data["slug"].strip()
             data["slug"] = data["slug"].replace(" ", "-")
             data["slug"] = re.sub(r"[^\w\-]+", "", data["slug"])
+        if "owners" in data:
+            data["owners"] = data.get("owners", [])
```

Comment:
Why are lines 95 - 96 required? If the `owners` key is in `data` then `data.get("owners", [])` is equivalent to `data["owners"]` and thus line 96 becomes `data["owners"] = data["owners"]` which is unnecessary.

Reply:
I think this is to make sure that `owners` is always a list if the key is present. With the help of these two rows, if `data == {"owners": None}`, `data` becomes `{"owners": []}`, but doesn't get set if `data` doesn't contain the key `owners`.

Reply:
@villebro per the [`dict.get`](https://docs.python.org/3/library/stdtypes.html#dict.get) docs, the default is only returned if he key does not exist. I think the desired logic should be, 

```python
if "owners" in data and data["owners"] is None:
  data["owners"] = []
```

Reply:
I agree @john-bodley , this is much more intuitive 👍

Reply:
yes, it's better, changed it